index.js

```diff
@@ -47,18 +47,45 @@ class LedgerBridgeKeyring extends EventEmitter {
     this.hdPath = opts.hdPath || hdPathString
     this.bridgeUrl = opts.bridgeUrl || BRIDGE_URL
     this.accounts = opts.accounts || []
-    this.accountIndexes = opts.accountIndexes || {}
+    this.accountDetails = opts.accountDetails || {}
+    if (!opts.accountDetails) {
+      this._migrateAccountDetails(opts)
+    }
+
     this.implementFullBIP44 = opts.implementFullBIP44 || false
 
-    if (this._isBIP44()) {
-      // Remove accounts that don't have corresponding account indexes
-      this.accounts = this.accounts
-        .filter((account) => Object.keys(this.accountIndexes).includes(ethUtil.toChecksumAddress(account)))
-    }
+    // Remove accounts that don't have corresponding account details
+    this.accounts = this.accounts
+      .filter((account) => Object.keys(this.accountDetails).includes(ethUtil.toChecksumAddress(account)))
 
     return Promise.resolve()
   }
 
+  _migrateAccountDetails (opts) {
+    if (this._isBIP44() && opts.accountIndexes) {
+      for (const account of Object.keys(opts.accountIndexes)) {
+        this.accountDetails[account] = {
+          bip44: true,
+          hdPath: this._getPathForIndex(opts.accountIndexes[account]),
+        }
+      }
+    }
+
+    // try to migrate non-bip44 accounts too
+    if (!this._isBIP44()) {
+      this.accounts.filter((account) => !Object.keys(this.accountDetails).includes(ethUtil.toChecksumAddress(account))).forEach((account) => {
+        try {
+          this.accountDetails[ethUtil.toChecksumAddress(account)] = {
+            bip44: false,
+            hdPath: this._pathFromAddress(account),
+          }
+        } catch (e) {
+          console.log(`failed to migrate account ${account}`)
+        }
+      })
```

Comment:
Nit: I found this a bit hard to read - I had to count parentheses to check whether the foreach was inside the filter or not.

Something like this would be better:

```suggestion
      this.accounts
        .filter((account) => !Object.keys(this.accountDetails).includes(ethUtil.toChecksumAddress(account)))
        .forEach((account) => {
          try {
            this.accountDetails[ethUtil.toChecksumAddress(account)] = {
              bip44: false,
              hdPath: this._pathFromAddress(account),
            }
          } catch (e) {
            console.log(`failed to migrate account ${account}`)
          }
      })
```